libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -759,6 +771,91 @@ bool BES_Select::removeSelectedItems() noexcept {
   }
 }
 
+void BES_Select::measureLengthInDirection(bool              directionBackwards,
+                                          const BI_NetLine& netline,
+                                          QSet<Uuid>&       visitedSegments,
+                                          UnsignedLength&   totalLength) {
+  auto currentNetline = std::cref(netline);
+  auto currentAnchor  = std::cref(directionBackwards ? netline.getStartPoint()
+                                                    : netline.getEndPoint());
+
+  for (;;) {
+    tl::optional<BI_NetLine&> nextNetline = tl::nullopt;
+    for (BI_NetLine* nl : currentAnchor.get().getNetLines()) {
+      // Don't visit a segment twice
+      if (visitedSegments.contains(nl->getUuid())) {
+        continue;
+      }
+      // Only visit selected segments
+      if (nl->isSelected()) {
+        if (nextNetline.has_value()) {
+          // There's already another connected and selected netline
+          throw LogicError(__FILE__, __LINE__,
+                           tr("Selected segment may not have branches"));
+        }
+
+        totalLength += nl->getLength();
+        nextNetline = *nl;
+        visitedSegments.insert(nl->getUuid());
+      }
+    }
+    if (nextNetline.has_value()) {
+      currentNetline = std::cref(*nextNetline);
+      currentAnchor =
+          std::cref(*nextNetline->getOtherPoint(currentAnchor.get()));
+    } else {
+      break;
+    }
+  }
+}
+
+bool BES_Select::measureSelectedItems(BI_NetLine& netline) noexcept {
+  Board* board = mEditor.getActiveBoard();
+  Q_ASSERT(board);
+  if (!board) return false;
+
+  Q_ASSERT(netline.isSelected());
+
+  // Store UUIDs of visited segments
+  QSet<Uuid> visitedSegments;
+  visitedSegments.insert(netline.getUuid());
+
+  // Get the netline length. Then traverse the selected segments first in one
+  // direction, then in the other direction.
+  UnsignedLength totalLength = netline.getLength();
+  try {
+    measureLengthInDirection(false, netline, visitedSegments,
+                             totalLength);  // can throw
+    measureLengthInDirection(true, netline, visitedSegments,
+                             totalLength);  // can throw
+  } catch (const LogicError& e) {
```

Comment:
I'd catch all exception types here since it's safer and has no disadvantages:

```suggestion
  } catch (const Exception& e) {
```

Reply:
In general I'd say that catching general exceptions is an antipattern (since specific exceptions might require specific handling), but in this case we're showing a generic error message anyways, so you're right :slightly_smiling_face: